### PR TITLE
[CIVisibility] - Add CIVisibility known errors to the CheckBuildLogs task

### PR DESCRIPTION
**Pull request by tonyredondo**
## Summary of changes

Fix CI

### CODE REVIEW

File: tracer/build/_build/Build.Steps.cs [hunk 1]
```diff
@@ -1882,6 +1882,11 @@ void RunWindowsIisIntegrationTests(Project project)
                new(@".*at CallTargetNativeTest\.NoOp\.Noop\dArgumentsVoidIntegration\.OnMethodBegin.*", RegexOptions.Compiled),
                new(@".*at CallTargetNativeTest\.NoOp\.Noop\dArgumentsVoidIntegration\.OnMethodEnd.*", RegexOptions.Compiled),
                new(@".*System.Threading.ThreadAbortException: Thread was being aborted\.", RegexOptions.Compiled),
+               // CI Visibility known errors
+               new (@".*The Git repository couldn't be automatically extracted.*", RegexOptions.Compiled),
+               new (@".*DD_GIT_REPOSITORY_URL is set with.*", RegexOptions.Compiled),
+               new (@".*The Git commit sha couldn't be automatically extracted.*", RegexOptions.Compiled),
+               new (@".*DD_GIT_COMMIT_SHA must be a full-length git SHA.*", RegexOptions.Compiled),
```
Comment (andrewlock) [inline tracer/build/_build/Build.Steps.cs:1889]: Do you definitely want to exclude these ones? We don't see them in CI currently, so not required I think?

**Review by andrewlock — APPROVED** [review]

File: tracer/build/_build/Build.Steps.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/build/_build/Build.Steps.cs:1889]: I know, I'm just playing safe here.

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3564 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`WriteAndFlushEnrichedTraces`|net472|780μs|452ns|1.75μs|0.386| 0|0|3.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|546μs|197ns|711ns|0| 0|0|2.63&nbsp;KB|
|#3564|`WriteAndFlushEnrichedTraces`|net472|760μs|279ns|1.01μs|0.374| 0|0|3.22&nbsp;KB|
|#3564|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|554μs|139ns|502ns|0| 0|0|2.63&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`AllCycleSimpleBody`|net472|25.2μs|128ns|556ns|0.277| 0|0|1.77&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`AllCycleSimpleBody`|netcoreapp3.1|23.1μs|133ns|1.04μs|0.0211| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`AllCycleMoreComplexBody`|net472|191μs|66ns|247ns|2.01| 0|0|13.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`AllCycleMoreComplexBody`|netcoreapp3.1|182μs|402ns|1.56μs|0.0918| 0|0|12.1&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`BodyExtractorSimpleBody`|net472|307ns|0.146ns|0.567ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`BodyExtractorSimpleBody`|netcoreapp3.1|251ns|0.519ns|1.94ns|0.00377| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`BodyExtractorMoreComplexBody`|net472|16μs|13.3ns|49.7ns|1.2| 0.0158|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.6μs|6.82ns|25.5ns|0.0879| 0|0|6.75&nbsp;KB|
|#3564|`AllCycleSimpleBody`|net472|25.1μs|38.4ns|149ns|0.278| 0|0|1.77&nbsp;KB|
|#3564|`AllCycleSimpleBody`|netcoreapp3.1|23μs|130ns|900ns|0.0118| 0|0|1.64&nbsp;KB|
|#3564|`AllCycleMoreComplexBody`|net472|194μs|92.6ns|359ns|2.02| 0|0|13.02&nbsp;KB|
|#3564|`AllCycleMoreComplexBody`|netcoreapp3.1|185μs|71.5ns|277ns|0.0922| 0|0|12.1&nbsp;KB|
|#3564|`BodyExtractorSimpleBody`|net472|278ns|0.253ns|0.945ns|0.0574| 0|0|361&nbsp;B|
|#3564|`BodyExtractorSimpleBody`|netcoreapp3.1|234ns|0.11ns|0.427ns|0.00366| 0|0|272&nbsp;B|
|#3564|`BodyExtractorMoreComplexBody`|net472|15.7μs|15ns|56.3ns|1.21| 0.0156|0|7.62&nbsp;KB|
|#3564|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.6μs|4.67ns|17.5ns|0.0884| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendRequest`|netcoreapp3.1|178μs|185ns|716ns|0.179| 0|0|20.39&nbsp;KB|
|#3564|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3564|`SendRequest`|netcoreapp3.1|179μs|191ns|738ns|0.267| 0|0|20.39&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`ExecuteNonQuery`|net472|1.83μs|0.971ns|3.63ns|0.16| 0.000931|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`ExecuteNonQuery`|netcoreapp3.1|1.47μs|0.405ns|1.52ns|0.014| 0|0|1&nbsp;KB|
|#3564|`ExecuteNonQuery`|net472|1.79μs|2.24ns|8.69ns|0.16| 0.000893|0|1.01&nbsp;KB|
|#3564|`ExecuteNonQuery`|netcoreapp3.1|1.45μs|1.65ns|6.4ns|0.0136| 0|0|1&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`CallElasticsearch`|net472|2.48μs|0.918ns|3.55ns|0.193| 0|0|1.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`CallElasticsearch`|netcoreapp3.1|1.51μs|0.514ns|1.99ns|0.0158| 0|0|1.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`CallElasticsearchAsync`|net472|2.6μs|1.74ns|6.74ns|0.214| 0|0|1.36&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`CallElasticsearchAsync`|netcoreapp3.1|1.68μs|0.377ns|1.46ns|0.0176| 0|0|1.28&nbsp;KB|
|#3564|`CallElasticsearch`|net472|2.52μs|2.75ns|10.7ns|0.193| 0|0|1.22&nbsp;KB|
|#3564|`CallElasticsearch`|netcoreapp3.1|1.53μs|0.885ns|3.19ns|0.0154| 0|0|1.16&nbsp;KB|
|#3564|`CallElasticsearchAsync`|net472|2.67μs|0.646ns|2.5ns|0.215| 0|0|1.36&nbsp;KB|
|#3564|`CallElasticsearchAsync`|netcoreapp3.1|1.66μs|0.93ns|3.6ns|0.0175| 0|0|1.28&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`ExecuteAsync`|net472|2.71μs|1.03ns|3.86ns|0.236| 0|0|1.49&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`ExecuteAsync`|netcoreapp3.1|1.77μs|0.479ns|1.86ns|0.0186| 0|0|1.41&nbsp;KB|
|#3564|`ExecuteAsync`|net472|2.61μs|1.37ns|5.14ns|0.236| 0|0|1.49&nbsp;KB|
|#3564|`ExecuteAsync`|netcoreapp3.1|1.78μs|1.37ns|5.3ns|0.0188| 0|0|1.41&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendAsync`|net472|5.74μs|2.53ns|9.81ns|0.449| 0|0|2.83&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendAsync`|netcoreapp3.1|3.63μs|1.85ns|6.93ns|0.0363| 0|0|2.66&nbsp;KB|
|#3564|`SendAsync`|net472|5.87μs|1.93ns|7.23ns|0.448| 0|0|2.83&nbsp;KB|
|#3564|`SendAsync`|netcoreapp3.1|3.6μs|1.17ns|4.53ns|0.0362| 0|0|2.66&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|net472|2.79μs|4.67ns|17.5ns|0.297| 0|0|1.88&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|netcoreapp3.1|2.37μs|0.818ns|2.95ns|0.0257| 0|0|1.91&nbsp;KB|
|#3564|`EnrichedLog`|net472|2.74μs|2.68ns|10ns|0.298| 0|0|1.88&nbsp;KB|
|#3564|`EnrichedLog`|netcoreapp3.1|2.28μs|1.41ns|5.46ns|0.0261| 0|0|1.91&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|net472|149μs|161ns|603ns|0.745| 0.223|0|4.72&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|netcoreapp3.1|121μs|189ns|733ns|0| 0|0|4.55&nbsp;KB|
|#3564|`EnrichedLog`|net472|148μs|122ns|474ns|0.742| 0.223|0|4.72&nbsp;KB|
|#3564|`EnrichedLog`|netcoreapp3.1|118μs|168ns|652ns|0| 0|0|4.55&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|net472|5.57μs|1.68ns|6.07ns|0.578| 0.00278|0|3.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|netcoreapp3.1|4.27μs|1.14ns|4.27ns|0.0538| 0|0|3.98&nbsp;KB|
|#3564|`EnrichedLog`|net472|5.49μs|2.38ns|9.2ns|0.579| 0.00273|0|3.65&nbsp;KB|
|#3564|`EnrichedLog`|netcoreapp3.1|4.16μs|2.08ns|7.77ns|0.0541| 0|0|3.98&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendReceive`|net472|2.21μs|3.45ns|13.4ns|0.228| 0|0|1.44&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`SendReceive`|netcoreapp3.1|1.77μs|0.794ns|3.07ns|0.0194| 0|0|1.38&nbsp;KB|
|#3564|`SendReceive`|net472|2.21μs|3.07ns|11.9ns|0.227| 0|0|1.44&nbsp;KB|
|#3564|`SendReceive`|netcoreapp3.1|1.69μs|0.846ns|3.17ns|0.0186| 0|0|1.38&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|net472|4.78μs|1.53ns|5.3ns|0.364| 0|0|2.3&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`EnrichedLog`|netcoreapp3.1|3.93μs|1.54ns|5.75ns|0.0255| 0|0|1.86&nbsp;KB|
|#3564|`EnrichedLog`|net472|4.76μs|1.31ns|4.9ns|0.365| 0|0|2.3&nbsp;KB|
|#3564|`EnrichedLog`|netcoreapp3.1|3.95μs|1.05ns|3.93ns|0.0236| 0|0|1.86&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`StartFinishSpan`|net472|1.12μs|1.55ns|5.79ns|0.139| 0|0|875&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`StartFinishSpan`|netcoreapp3.1|904ns|0.381ns|1.47ns|0.0112| 0|0|824&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`StartFinishScope`|net472|1.35μs|0.622ns|2.33ns|0.151| 0|0|955&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`StartFinishScope`|netcoreapp3.1|1.07μs|0.454ns|1.7ns|0.0123| 0|0|944&nbsp;B|
|#3564|`StartFinishSpan`|net472|1.18μs|0.554ns|2.07ns|0.139| 0|0|875&nbsp;B|
|#3564|`StartFinishSpan`|netcoreapp3.1|935ns|0.29ns|1.09ns|0.0112| 0|0|824&nbsp;B|
|#3564|`StartFinishScope`|net472|1.42μs|0.633ns|2.45ns|0.151| 0|0|955&nbsp;B|
|#3564|`StartFinishScope`|netcoreapp3.1|1.11μs|0.752ns|2.91ns|0.0127| 0|0|944&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`RunOnMethodBegin`|net472|1.47μs|0.436ns|1.69ns|0.152| 0|0|955&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/32ca48e0262d192552df5d3d3a6e3665244ef03c)|`RunOnMethodBegin`|netcoreapp3.1|1.13μs|0.428ns|1.6ns|0.013| 0|0|944&nbsp;B|
|#3564|`RunOnMethodBegin`|net472|1.47μs|0.442ns|1.71ns|0.151| 0|0|955&nbsp;B|
|#3564|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|0.392ns|1.52ns|0.0127| 0|0|944&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3564 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4a392fd6f995fd9312a27d99a00d50cb8f3d98f) will **not change** line coverage
:warning: Merging #3564 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4a392fd6f995fd9312a27d99a00d50cb8f3d98f) will will **decrease** branch coverage by `1%`
:no_entry: Merging #3564 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4a392fd6f995fd9312a27d99a00d50cb8f3d98f) will will **increase** complexity by `561`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4a392fd6f995fd9312a27d99a00d50cb8f3d98f) | #3564       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `20114` / `28304` | `20554` / `28932` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `12029` / `18069` | `12277` / `18596` |          |
| Branches %| `67%`      | `66%`      |  `-1%` :warning:  |
| Complexity|   `19496`      | `20057`        |  `561`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=121681&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/121681/artifacts?artifactName=Code%20Coverage%20Report_121681&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3564](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=122173&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/122173/artifacts?artifactName=Code%20Coverage%20Report_122173&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4a392fd6f995fd9312a27d99a00d50cb8f3d98f) | #3564       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `67%`      | `66%`       |  `-1%` :warning:  |
| Complexity| `19496`      | `20057`       |  `561` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Iast.IastRequestContext](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/IAST/IastRequestContext.cs) | `-10%` :no_entry: | `-8%` :no_entry:   | `24` :no_entry: |
| [Datadog.Trace.Logging.DatadogLogging](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs) | `-6%` :no_entry: | `-23%` :no_entry:   | `6` :no_entry: |
| [Datadog.Trace.AppSec.ControllerContextExtensions](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/AppSec/ControllerContextExtensions.Framework.cs) | `-5%` :no_entry: | `-37%` :no_entry:   | `16` :no_entry: |
| [Datadog.Trace.Propagators.W3CTraceContextPropagator](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs) | `-2%` :warning: | `-13%` :no_entry:   | `173` :no_entry: |
| [Datadog.Trace.AspNet.TracingHttpModule](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs) | `-1%` :warning: | `-11%` :no_entry:   | `16` :no_entry: |
| [Datadog.Trace.SpanExtensions](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/SpanExtensions.cs) | `7%` :heavy_check_mark: | `5%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ExtensionMethods.SpanExtensions](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs) | `8%` :heavy_check_mark: | `11%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `14%` :heavy_check_mark: | `11%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Coverage.CoverageReporter](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/Ci/Coverage/CoverageReporter.cs) | `21%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `6` :no_entry: |
| [Datadog.Trace.Ci.Coverage.CoverageEventHandler](https://github.com/DataDog/dd-trace-dotnet/tree/16e0ba1f1602e8dc853b110af6e5c4b1b7523c4d/tracer/src/Datadog.Trace/Ci/Coverage/CoverageEventHandler.cs) | `31%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `-5` :heavy_check_mark: |
| ...And 2 more  | | | |


The following classes were added in #3564:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Ci.Coverage.DefaultWithGlobalCoverageEventHandler](https://github.com/DataDog/dd-trace-dotnet/pull/3564/files) | `22%` | `16%` | `78` |
| [Datadog.Trace.Ci.Coverage.Models.Global.ComponentCoverageInfo](https://github.com/DataDog/dd-trace-dotnet/pull/3564/files) | `0%` | `0%` | `29` |
| [Datadog.Trace.Ci.Coverage.Models.Global.CoverageInfo](https://github.com/DataDog/dd-trace-dotnet/pull/3564/files) | `48%` | `48%` | `44` |
| [Datadog.Trace.Ci.Coverage.Models.Global.FileCoverageInfo](https://github.com/DataDog/dd-trace-dotnet/pull/3564/files) | `0%` | `0%` | `31` |
| [Datadog.Trace.Ci.Coverage.Models.Global.GlobalCoverageInfo](https://github.com/DataDog/dd-trace-dotnet/pull/3564/files) | `6%` | `0%` | `29` |
| ...And 4 more  | | | |


1 classes were removed from Datadog.Trace in #3564


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=121681&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/121681/artifacts?artifactName=Code%20Coverage%20Report_121681&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3564](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=122173&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/122173/artifacts?artifactName=Code%20Coverage%20Report_122173&api-version=6.0&%24format=zip)

**Pull request merged**